x/oracle/keeper/keeper.go

```diff
@@ -200,21 +184,21 @@ func (k Keeper) ProcessUpdateWhiteListValidator(ctx sdk.Context, cosmosSender sd
 // power to be considered successful, or alternatively,
 // will never be able to become successful due to not enough validation power being
 // left to push it over the threshold required for consensus.
-func (k Keeper) processCompletion(ctx sdk.Context, prophecy types.Prophecy) types.Prophecy {
-	highestClaim, highestClaimPower, totalClaimsPower, totalPower := prophecy.FindHighestClaim(ctx, k.stakeKeeper, k.GetOracleWhiteList(ctx))
+func (k Keeper) processCompletion(ctx sdk.Context, networkDescriptor types.NetworkIdentity, prophecy types.Prophecy) types.Prophecy {
+	validators := k.GetOracleWhiteList(ctx, networkDescriptor)
+	fmt.Printf("processCompletion %v \n", validators)
 
-	highestConsensusRatio := float64(highestClaimPower) / float64(totalPower)
-	remainingPossibleClaimPower := totalPower - totalClaimsPower
-	highestPossibleClaimPower := highestClaimPower + remainingPossibleClaimPower
-	highestPossibleConsensusRatio := float64(highestPossibleClaimPower) / float64(totalPower)
-
-	if highestConsensusRatio >= k.consensusNeeded {
+	highestClaim, highestRatio, vetoRatio := validators.GetPowerRatio(prophecy.ClaimValidators)
+	fmt.Printf("processCompletion %s, %f, %f \n", highestClaim, highestRatio, vetoRatio)
+	if highestRatio >= k.consensusNeeded {
 		prophecy.Status.Text = types.StatusText_STATUS_TEXT_SUCCESS
 		prophecy.Status.FinalClaim = highestClaim
-	} else if highestPossibleConsensusRatio < k.consensusNeeded {
+	} else if vetoRatio > (1.0 - k.consensusNeeded) {
```

Comment:
Veto's should not even be a thing. A validator can either approve a prophecy claim, or they can not approve it. Vetoing should not be allowed.

Reply:
will be removed after new prophecy.